src/server/lib/okta/tokens.ts

```diff
@@ -0,0 +1,167 @@
+import { Request, CookieOptions } from 'express';
+import { ResponseWithRequestState } from '@/server/models/Express';
+import OktaJwtVerifier from '@okta/jwt-verifier';
+import { joinUrl } from '@guardian/libs';
+import { getConfiguration } from '@/server/lib/getConfiguration';
+import { logger } from '@/server/lib/serverSideLogger';
+import { GU_DOMAIN } from '@/server/models/Configuration';
+
+/**
+ * Configuration
+ */
+const { okta, baseUri, stage } = getConfiguration();
+
+/**
+ * OAuth Token Name Type
+ */
+type OAuthCookieNames =
+  | typeof OAuthAccessTokenCookieName
+  | typeof OAuthIdTokenCookieName;
+
+/**
+ * Name of the OAuth Access Token cookie to set
+ * @type {string}
+ */
+const OAuthAccessTokenCookieName = 'GU_ACCESS_TOKEN';
+
+/**
+ * Name of the OAuth ID Token cookie to set
+ * @type {string}
+ */
+const OAuthIdTokenCookieName = 'GU_ID_TOKEN';
+
+const issuerDomain = () => {
+  // if we're in cypress mocked, we need to add https to the issuer domain, just to get it
+  // to validate, even if we never actually use the verifier in cypress mocked tests
+  if (process.env.RUNNING_IN_CYPRESS_MOCKED === 'true') {
+    return okta.orgUrl.replace('http://', 'https://');
```

Comment:
Why does `okta.orgUrl` have 'http' in it in the first place? Is that for local testing?

Reply:
yeah, its for mocked local testing, where we sometimes need to check if we've redirected to the oauth flow using the mocked server, or return a generic response to test specific error cases